### ISSUE TITLE
Finetuning model: Freeze layer or Mixout transfer learning.

### ISSUE THREAD

**Issue by Rachine** [pyannote/pyannote-audio#428]
Hello,

I am trying to do two things:
1) I am trying to fine-tune speaker embedding from VoxCeleb pre-trained model on a smaller dataset of around 60 speakers. 
I wanted to try to freeze some layer in the SincTDNN model:
I tried to do some 'hack'. I loaded the model from torch hub:
`/path/to/hub/models/emb_voxceleb/train/X.SpeakerDiarization.VoxCeleb.train/weights/0326.pt`

After loading the model, i wanted to freeze the gradient to the first layers with the setting param.requires_grad to false and resaved the model as pt with the method:

`torch.save(model, NEW_PATH)` not with `torch.save(net.state_dict(), NEW_PATH)`

Otherwise the param.requires_grad is not saved along with the model. 

Yet, when i specified the new path with :
`pyannote-audio emb train --pretrained=NEW_PATH--subset=train --to=100 --parallel=8 ${EXP_DIR} MyProtocol`
I got the error, 

```
Could not assign model weights. The following exception was raised:

Can't get attribute 'SincTDNN' on <module '__main__' from '/home/rriad/.conda/envs/pyannote-ezh/bin/pyannote-audio'>

Are you sure the architectures are consistent?

```
It is also happening when I am saving a model without any modifications. 
Do you have any advice how I could do it? 

2) I also want to try to fine tune a model with `mixout`, a model can be wrapped with this function pytorch: 
https://gist.github.com/Rachine/7d6d72adcc0b38c4a4db68491dc26797#file-mixout-py-L25
The pre-trained model needs to be wrapped `model.apply(MixoutWrapper)`. 

The hook function and all can not be saved, so it would be to apply this wrapper to a pre-trained model before retraining on the smaller dataset I assume.

Do you have any advice where to insert this apply mode in the code base?

Thank you very much! 

Cheers :)

**Comment by Rachine** [comment]
I just saw this commit, it seems to be that the layer freezer is implemented somehow. I need to figure out how to use it now https://github.com/pyannote/pyannote-audio/commit/feb19bcc41351eac31d96fbb86fbf08940d51427

**Comment by hbredin** [comment]
I did implement freezing some time ago but never really tested it. 
I'd love to get feedback about this feature.

This is done by adding a `freeze:` section in the configuration file.
For instance, this should freeze `sincnet` feature extractor`tdnn` layers and first embedding layer of `SincTDNN`.

```yaml
freeze:
    - sincnet_
    - tdnn_.tdnn
    - tdnn_.segment6
```

To answer your question about `mixout`, I would need to spend more time to understand what the code does (time which I unfortunately do not have right now...)
